packages/grid/_modules_/grid/hooks/features/useGridBaseComponentProps.tsx

```diff
@@ -1,28 +1,28 @@
 import * as React from 'react';
-import { GridApiRef } from '../../models/api/gridApiRef';
+import { GridApiContext } from '../../components/GridApiContext';
 import { GridBaseComponentProps } from '../../models/params/gridBaseComponentProps';
 import { optionsSelector } from '../utils/optionsSelector';
 import { visibleGridColumnsSelector } from './columns/gridColumnsSelector';
 import { useGridSelector } from './core/useGridSelector';
 import { useGridState } from './core/useGridState';
 import { unorderedGridRowModelsSelector } from './rows/gridRowsSelector';
 
-export const useGridBaseComponentProps = (apiRef: GridApiRef | undefined) => {
+export const useGridBaseComponentProps = () => {
+  const apiRef = React.useContext(GridApiContext);
   const options = useGridSelector(apiRef, optionsSelector);
   const rows = useGridSelector(apiRef, unorderedGridRowModelsSelector);
   const columns = useGridSelector(apiRef, visibleGridColumnsSelector);
   const [state] = useGridState(apiRef!);
 
-  const baseComponentProps: GridBaseComponentProps | undefined = React.useMemo(
-    () =>
-      apiRef && {
-        state,
-        rows,
-        columns,
-        options,
-        api: apiRef,
-        rootElement: apiRef.current.rootElementRef!,
-      },
+  const baseComponentProps: GridBaseComponentProps = React.useMemo(
+    () => ({
+      state,
+      rows,
+      columns,
+      options,
+      api: apiRef!,
```

Comment:
I put api as I was returning apiRef.current. If you return apiRef, you might want to rename that and put the type

Reply:
Could you be more specific. I didn't change what was assigned to `api`, it was always `api: apiRef`. Also, the type coming from `GridBaseComponentProps` is correct.

Reply:
How about we rename it to apiRef then?

Reply:
Updated